version: '3.8'

services:

  nginx:
    image: nginx:alpine
    container_name: nginx
    ports:
      - "80:80"
    volumes:
      - ./docker/nginx/default.conf:/etc/nginx/conf.d/default.conf
      - ./public:/var/www/html/public
    depends_on:
      - app
      - db
    networks:
      - fis-network

  db:
    image: mysql:8.0
    container_name: mysql
    restart: unless-stopped
    volumes:
      - dbdata:/var/lib/mysql
    environment:
      MYSQL_DATABASE: ${DB_DATABASE}
      MYSQL_ROOT_PASSWORD: ${DB_PASSWORD}
      MYSQL_PASSWORD: ${DB_PASSWORD}
    ports:
      - "3306:3306"
    networks:
      - fis-network

  app:
    container_name: php
    build:
      args:
        user: milos
        uid: 1000
      context: .
      dockerfile: ./docker/php/dev/Dockerfile
    volumes:
      - ./:/var/www/html
#      - /var/www/html/vendor
      - ./docker/php/dev/conf.d/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
      - ./docker/php/dev/conf.d/error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
    expose:
      - "9000"
    depends_on:
      - db
    networks:
      - fis-network


volumes:
  dbdata:
      driver: local

networks:
  fis-network:
    driver: bridge

